packages/hub/src/hooks/useAvailableHeight.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react";
+
+//This gets all of the available height on the page, starting from the top of the element.
+//This is useful for having a div fill up the rest of the page.
+export const useAvailableHeight = () => {
+  const ref = useRef<HTMLDivElement>(null);
+  const [height, setHeight] = useState(0);
+
+  useEffect(() => {
+    const updateHeight = () => {
+      if (ref.current) {
+        setHeight(window.innerHeight - ref.current.offsetTop);
+      }
+    };
+
+    // Call it right away
+    updateHeight();
+
+    // Call it when window resizes
+    window.addEventListener("resize", updateHeight);
```

Comment:
I feel like this would be a bit much (bad performance/experience). My guess is that we should find some way to remove the footer on this page, in which case there shouldn't be scroll events. 

Are there other examples you know of, of apps doing this?

Reply:
I asked GPT4, here was it's response. Not something to be taken incredibly seriously, but something.

<img width="930" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/e07a29f0-8da7-40e0-979e-a274e39714d9">